filebeat/docs/inputs/input-winlog.asciidoc

```diff
@@ -0,0 +1,476 @@
+:type: winlog
+
+[id="{beatname_lc}-input-{type}"]
+=== winlog input
+
+++++
+<titleabbrev>winlog</titleabbrev>
+++++
+
+beta[]
+
+Use the `winlog` input to read Windows event logs. It reads from one or more 
+event logs using Windows APIs, filters the events based on user-configured criteria, 
+then sends the event data to the configured outputs. It watches the event
+logs so that new event data is sent in a timely manner. The read position for
+each event log is persisted to disk to allow the input to resume after
+restarts.
+
+The `winlog` input can capture event data from any event logs running on your system.
+For example, you can capture events such as:
+
+* application events
+* hardware events
+* security events
+* system events
+
+Here is a sample configuration:
+
+[source,yaml]
+--------------------------------------------------------------------------------
+- type: winlog
```

Comment:
Are there benefits to this config format beyond being able to paste in Winlogbeat config? 

Personally I like the one input to one reader format for its simplicity. The configuration of `id` at two different levels is confusing. e.g.

```yaml
filebeat.inputs:
- type: winlog
  id: winlog-abc
  event_logs:
  - id: winlog-xml-query-123
    xml_query: >
      <QueryList>
        <Query Id="0" Path="DhcpAdminEvents">
          <Select Path="DhcpAdminEvents">*</Select>
          <Select Path="Microsoft-Windows-Dhcp-Server/FilterNotifications">*</Select>
          <Select Path="Microsoft-Windows-Dhcp-Server/Operational">*</Select>
        </Query>
      </QueryList>
```

Reply:
The intent was exactly that, to be able to paste winlogbeat configs directly. I agree that it might cause some confusion with the IDs. I do not have strong feelings against moving towards current integrations format where the input only holds one event log config. Even though some requests such as https://github.com/elastic/integrations/issues/5877 would indicate both have its benefits (I guess we could implement this in a different way on agent side, also).

Reply:
I would prefer to have only one way to configure the input, so can we take the route of requiring one input entry per channel?

Regarding https://github.com/elastic/integrations/issues/5877, I think this is addressable in such a way that Fleet produces multiple `streams` in the same package policy. It requires us to control the UI for the package (similar to how the "Defend for Containers" integration UI works), but I think this is needed anyways to provide a good UX. For reference, this is the agent policy for winlog now. I'm assuming we can populate more than one stream (or more than one input) and assign each unique ID. If I'm wrong, then worst case is we go back on this decision and accept `event_logs`.



```yaml
inputs:
  - id: winlog-winlogs-db8ca3a1-19a7-4be1-91e9-274955806d05
    name: winlog-1
    revision: 1
    type: winlog
    use_output: default
    meta:
      package:
        name: winlog
        version: 2.1.2
    data_stream:
      namespace: default
    package_policy_id: db8ca3a1-19a7-4be1-91e9-274955806d05
    streams:
      - id: winlog-winlog.winlogs-db8ca3a1-19a7-4be1-91e9-274955806d05
        name: Security
        data_stream:
          dataset: winlog.winlog
        condition: '${host.platform} == ''windows'''
        ignore_older: 72h
```

For completeness I'll mention that a [Transformation](https://github.com/elastic/beats/blob/e177fc500b53163b8fdb62a9069dcf0d07b304dc/x-pack/libbeat/management/devguide.asciidoc#config-transformation) could be used, but I don't really like the idea of this because its like an invisible layer between the beat config and agent. Packetbeat and `network_traffic` use this, and its confusing to see agent configs that don't match with Packetbeat docs.